include/ghex/transport_layer/ucx/request.hpp

```diff
@@ -100,27 +93,24 @@ namespace gridtools{
                     /** @brief return pointer to an instance from ucx provided storage pointer */
                     static request_data_cb& get(void* __restrict ptr)
                     {
+                        static constexpr std::uintptr_t mask = ~(alignof(request_data_cb)-1u);
                         unsigned char* __restrict cptr = (unsigned char*)ptr;
                         return *reinterpret_cast<request_data_cb*>
                         ((reinterpret_cast<std::uintptr_t>(cptr) + alignof(request_data_cb)-1) & mask);
                     }
                 };
-                using request_data_size_cb = std::integral_constant<std::size_t,
-                sizeof(request_data_cb<::gridtools::ghex::threads::atomic::primitives>) +
-                alignof(request_data_cb<::gridtools::ghex::threads::atomic::primitives>)>;
 
                 /** @brief minimum size of the ucx provided storage in bytes */
-                using request_data_size = request_data_size_cb;
-                
+                using request_data_size = std::integral_constant<std::size_t,sizeof(request_data_cb)>;
```

Comment:
why did you remove the alignment padding? Not sure if this is safe since we have no idea what size the ucx will use for its own request.

Reply:
They were not used anyway, IIRC

Reply:
see https://github.com/GridTools/GHEX/blob/6c04d22c30a22956e7cc079819360e253535ac8f/include/ghex/transport_layer/ucx/context.hpp#L130

Reply:
I re-introduced the alignment (hoping that this would fix the bug, but it did not)